### PR TITLE
Update the SDK to use Microsoft Accounts authentication

### PR DESCRIPTION
**Pull request by bfreydier**

### CODE REVIEW

File: src/GrooveApiClient/GrooveClient.cs [hunk 1]
```diff
@@ -1,566 +1,567 @@
...
+            }
+
+            return requestParameters;
+        }
+
+        private async Task<Dictionary<string, string>> FormatRequestHeadersAsync(string userAuthorizationHeader)
+        {
+            MicrosoftAccountAuthenticationCache.AccessToken token =
+                await _microsoftAccountAuthenticationCache.CheckAndRenewTokenAsync(new CancellationToken(false));
+
```
Comment (antoinecellerier) [inline src/GrooveApiClient/GrooveClient.cs]: This is only needed if the request is *not* user authenticated. If you already have a user authentication header, you should skip all of this #Resolved

File: src/GrooveApiDataContract/ErrorCode.cs [hunk 2]
```diff
@@ -123,27 +123,27 @@ public enum ErrorCode
         #endregion
 
         #region AccessToken errors
-        [Description("Azure Marketplace access token required")]
+        [Description("Microsoft access token required")]
```
Comment (antoinecellerier) [inline src/GrooveApiDataContract/ErrorCode.cs]: >Microsoft [](start = 22, length = 9)

I'm not sure that Microsoft access token or Microsoft token are things which exist. We should use the same words as those used in the documentation for consistency. ("application access token" & "user access token" I guess) #Resolved

**Comment by antoinecellerier** [comment]
:shipit:

File: src/GrooveApiClient/GrooveClient.cs (hunk 1 above)
Comment (bfreydier) [inline src/GrooveApiClient/GrooveClient.cs]: This version of the sdk does not support user auth

---
In reply to: [94612088](https://github.com/Microsoft/groove-api-sdk-csharp/pull/1#discussion_r94612088) [](ancestors = 94612088)

File: src/GrooveApiClient/MicrosoftAccountAuthenticationCache.cs [hunk 3]
```diff
@@ -1,78 +1,72 @@
...
+namespace Microsoft.Groove.Api.Client
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.Groove.Api.DataContract.AuthenticationDataContract;
+
+    /// <summary>
+    /// Basic Microsoft Account Application authenticationauthentication cache
```
Comment (daauro) [inline src/GrooveApiClient/MicrosoftAccountAuthenticationCache.cs]: >authenticationauthentication [](start = 44, length = 28)

authenticathenticauthenticationtication ? #Resolved

File: src/GrooveApiDataContract/ErrorCode.cs [hunk 4]
```diff
@@ -123,27 +123,27 @@ public enum ErrorCode
         #endregion
 
         #region AccessToken errors
-        [Description("Azure Marketplace access token required")]
+        [Description("Application access token required")]
```
Comment (daauro) [inline src/GrooveApiDataContract/ErrorCode.cs:126]: >Application [](start = 22, length = 11)

were those renames done in the service too? #Resolved

**Comment by daauro** [comment]
:shipit:

File: src/GrooveApiDataContract/ErrorCode.cs (hunk 4 above)
Comment (bfreydier) [reply src/GrooveApiDataContract/ErrorCode.cs:126]: yup (in review)

---
In reply to: [94793263](https://github.com/Microsoft/groove-api-sdk-csharp/pull/1#discussion_r94793263) [](ancestors = 94793263)

**Pull request merged**